x-pack/plugins/infra/server/routes/infra/lib/mapper.ts

```diff
@@ -32,16 +34,29 @@ export const mapToApiResponse = (
     };
   }
 
-  const hosts = buckets.map((bucket) => {
-    const metrics = convertMetricBucket(params, bucket);
-    const metadata = convertMetadataBucket(bucket);
+  const hosts = buckets
+    .map((bucket) => {
+      const metrics = convertMetricBucket(params, bucket);
+      const metadata = convertMetadataBucket(bucket);
 
-    return { name: bucket.key as string, metrics, metadata };
-  });
+      return { name: bucket.key as string, metrics, metadata };
+    })
+    .sort((a, b) => {
+      const cpuAValue = a.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
+      const cpuBValue = b.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
```

Comment:
Can't you already initialize cpuAValue and cpuBValue here?
```suggestion
      const cpuAValue = a.metrics.find((metric) => metric.name === 'cpu')?.value ?? 0;
      const cpuBValue = b.metrics.find((metric) => metric.name === 'cpu')?.value ?? 0;
```

Reply:
>I didn't dare to do it as I wanted to keep the response of the metrics as it is, so if the value is null keep it null, don't want to mess up with how the code was already done too much

But this is just the sorting function, setting the cpuAValue to 0 won't change the value of the obj itself.